system-addon/bootstrap.js

```diff
@@ -0,0 +1,91 @@
+/* globals Components, XPCOMUtils, Preferences, ActivityStream */
+"use strict";
+
+const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Preferences.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "ActivityStream",
+  "resource://activity-stream/lib/ActivityStream.jsm");
+
+const ACTIVITY_STREAM_ENABLED_PREF = "browser.newtabpage.activity-stream.enabled";
+const ACTIVITY_STREAM_OPTIONS = {newTabURL: "resource://activity-stream/data/activity-stream.html"};
+const REASON_STARTUP_ON_PREF_CHANGE = "PREF_ON";
+const REASON_SHUTDOWN_ON_PREF_CHANGE = "PREF_OFF";
+
+let activityStream;
+let startupData;
+
+/**
+ * init - Initializes an instance of ActivityStream. This could be called by
+ *        the startup() function exposed by bootstrap.js, or it could be called
+ *        when ACTIVITY_STREAM_ENABLED_PREF is changed from false to true.
+ *
+ * @param  {string} reason - Reason for initialization. Could be install, upgrade, or pref change
+ */
```

Comment:
If I'm reading the code right, reason could also be PREF_ON or PREF_OFF.  Probably worth documenting here.